.github/workflows/first_contrib_cert_generator.yml

```diff
@@ -0,0 +1,270 @@
+          PR_NUMBER="${{ github.event.inputs.pr_number || github.event.pull_request.number }}"
+          MERGE_DATE=$(date -u +"%B %d, %Y")
+
+          sed -i "/id=\"contributorName\"/s/value=\"[^\"]*\"/value=\"${CONTRIBUTOR_NAME}\"/" ${HTML_FILE} || { echo "ERROR: Failed to update contributor name."; exit 1; }
+          sed -i "/id=\"prNumber\"/s/value=\"[^\"]*\"/value=\"#${PR_NUMBER}\"/" ${HTML_FILE} || { echo "ERROR: Failed to update PR number."; exit 1; }
+          sed -i "/id=\"mergeDate\"/s/value=\"[^\"]*\"/value=\"${MERGE_DATE}\"/" ${HTML_FILE} || { echo "ERROR: Failed to update merge date."; exit 1; }
+
+      # Step 4: Setup Node.js environment
+      - name: Setup Node.js
+        if: ${{ github.event_name == 'workflow_dispatch' || steps.check_first_pr.outputs.is_first_pr == 'true' }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18'
+
+      # Step 5: Install Puppeteer
+      - name: Install Puppeteer
+        if: ${{ github.event_name == 'workflow_dispatch' || steps.check_first_pr.outputs.is_first_pr == 'true' }}
+        run: |
+          npm install puppeteer || { echo "ERROR: Failed to install Puppeteer."; exit 1; }
+
+      # Step 6: Take a screenshot of the certificate div
+      - name: Create and run screenshot script
+        if: ${{ github.event_name == 'workflow_dispatch' || steps.check_first_pr.outputs.is_first_pr == 'true' }}
+        run: |
+          cat <<'EOF' > screenshot.js
+          const puppeteer = require('puppeteer');
+          const path = require('path');
+          (async () => {
+            const browser = await puppeteer.launch({ args: ['--no-sandbox', '--disable-setuid-sandbox'] });
+            const page = await browser.newPage();
+            const htmlPath = 'file://' + path.resolve('automation-repo/certificate-generator/certificate_generator.html');
+            await page.goto(htmlPath, { waitUntil: 'networkidle0' });
+            await page.setViewport({ width: 1080, height: 720 });
+            const element = await page.$('#certificatePreview');
+            if (!element) {
+              console.error('Could not find element #certificatePreview.');
+              process.exit(1);
+            }
+            await element.screenshot({ path: 'certificate.png' });
+            await browser.close();
+            console.log('Screenshot saved as certificate.png');
+          })().catch(err => {
+            console.error(err);
+            process.exit(1);
+          });
+          EOF
+          node screenshot.js || { echo "ERROR: Screenshot script failed."; exit 1; }
+
+      # Step 7: Upload certificate image to separate repository
+      - name: Upload certificate to separate repository
+        if: ${{ github.event_name == 'workflow_dispatch' || steps.check_first_pr.outputs.is_first_pr == 'true' }}
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.CERTIFICATES_REPO_TOKEN || github.token }}
```

Comment:
the fallback to `github.token` seems incorrect? There is no way the default action token would work to upload to a different repo.

Reply:
Thats a debug part that I left out, good catch, removed the fallback part.